src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs

```diff
@@ -0,0 +1,174 @@
+                initializers.Add(new ObjectPropertyInitializer(autoRestProperty, reference));
+            }
+
+            ObjectTypeConstructor ctor = new ObjectTypeConstructor(memberDeclarationOptions, parameters.ToArray(), initializers.ToArray(), GetBaseCtor());
+            return ctor;
+        }
+
+        protected override IEnumerable<ObjectTypeProperty> BuildProperties()
+        {
+            foreach (var property in _myProperties)
+            {
+                var getter = property.GetGetMethod();
+                var setter = property.GetSetMethod();
+                MemberDeclarationOptions memberDeclarationOptions = new MemberDeclarationOptions(
+                    getter is not null && getter.IsPublic ? "public" : "internal",
+                    property.Name,
+                    GetTypeFromSystem(property.PropertyType));
+                Property prop = new Property();
+                prop.Nullable = false;
+                prop.ReadOnly = GetReadOnly(property); //TODO read this from attribute from reference object
+                prop.SerializedName = ToCamelCase(property.Name);
+                prop.Summary = $"Gets{GetPropertySummary(setter)} {property.Name}";
+                prop.Required = true;
+
+                //We are only handling a small subset of cases because the set of reference types used from Azure.ResourceManager.Core is known
+                //If in the future we add more types which have unique cases we might need to update this code, but it will be obvious
+                //given that the generation will fail with the new types
+                if (property.PropertyType.IsGenericType && property.PropertyType.GetGenericTypeDefinition() == typeof(IDictionary<,>))
+                {
+                    prop.Schema = new DictionarySchema();
+                    prop.Schema.Type = AllSchemaTypes.Dictionary;
+                }
+
+                yield return new ObjectTypeProperty(memberDeclarationOptions, prop.Summary, prop.IsReadOnly, prop, GetTypeFromSystem(property.PropertyType));
+            }
+        }
+
+        private bool GetReadOnly(PropertyInfo property)
+        {
+            if (property.Name == "Tags")
+                return false;
+            return property.GetSetMethod() is null;
+        }
+
+        private CSharpType GetTypeFromSystem(Type type)
+        {
+            List<CSharpType> args = new List<CSharpType>();
+            if (type.IsGenericType)
+            {
+                foreach (var argType in type.GetGenericArguments())
+                {
+                    args.Add(GetTypeFromSystem(argType));
+                }
+            }
+            return new CSharpType(type, false, args.ToArray());
+        }
+
+        private string GetPropertySummary(MethodInfo? setter)
+        {
+            return setter is not null ? " or sets" : string.Empty;
+        }
+
+        protected override ObjectTypeConstructor? BuildSerializationConstructor() => null;
```

Comment:
I have a suspicion that everything that's not implemented on the SystemObjectType can be moved from ObjectType to SchemaObjectType

Reply:
this has an implementation now

Reply:
The implementation is to return empty array. I still don't see why it couldn't move to SchemaObjectType?

Reply:
it doesn't return an empty array, and why the ObjectType needs the definition is explained here https://github.com/Azure/autorest.csharp/pull/1170#discussion_r618908466